vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt

```diff
@@ -83,20 +83,17 @@ class AndroidDevice(val ddmsDevice: IDevice,
      * We can only call this after the device finished booting
      */
     private val realSerialNumber: String by lazy {
-        val marathonSerialProp: String = ddmsDevice.getProperty("marathon.serialno") ?: ""
         val serialProp: String = ddmsDevice.getProperty("ro.boot.serialno") ?: ""
         val hostName: String = ddmsDevice.getProperty("net.hostname") ?: ""
         val serialNumber = ddmsDevice.serialNumber
 
         val result = when (serialStrategy) {
             SerialStrategy.AUTOMATIC -> {
-                marathonSerialProp.takeIf { it.isNotEmpty() }
```

Comment:
See https://github.com/Malinskiy/marathon/issues/223#issuecomment-491481014 for why this should be here. Marathon serial property was implemented for agoda's device farm

Reply:
Well, there is no such property on emulators - most probably this feature is broken in adb-butler. So I needed a quick fix for new emulators, that don't have any other id.

Reply:
There was one: https://github.com/agoda-com/adb-butler/commit/1171a5f459d84e958f9f3c9b195208b5ee89e929